indico/web/client/js/react/components/IButton.tsx

```diff
@@ -0,0 +1,79 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React from 'react';
+
+import {toClasses} from 'indico/react/util';
+
+interface IButtonPropTypes {
+  classes: object;
+  href: string | undefined;
+  title: string;
+  children: React.ReactNode;
+  onClick: (() => undefined) | undefined;
+  highlight: boolean;
+  disabled: boolean;
+  icon: string;
+  dropdown: boolean;
+  borderless: boolean;
+  style: object;
+}
+
+/** Legacy-style Indico UI Button */
+export default function IButton({
+  classes = {},
+  href,
+  title,
+  children,
+  onClick,
+  highlight = false,
+  disabled = false,
+  icon = '',
+  dropdown = false,
+  borderless = false,
+  style = {},
+}: IButtonPropTypes) {
+  const finalClasses: Record<string, boolean> = {
+    ...classes,
+    'i-button': true,
+    disabled,
+    highlight,
+    borderless,
+  };
+
+  if (icon) {
+    finalClasses[`icon-${icon}`] = true;
+  }
+
+  if (dropdown) {
+    finalClasses['arrow'] = true;
+  }
+
+  const attrs: Record<string, string | object | (() => undefined)> = {
```

Comment:
there's no generic "dict" type? Having to provide such a verbose type definition for something purely internal feels like being unnecessarily verbose...

Reply:
In general, dictionaries are a bit complicated in TS. In this specific case you could just use a simpler type e.g. `Record<string, any>` or `{[key: string]: any}`. The reason you can't leave out the type altogether is the assignment that follows: `attrs['onClick'] = onClick` which requires that `attrs` has string keys.